worlds/inscryption/__init__.py

```diff
@@ -0,0 +1,126 @@
+from BaseClasses import Region, Item, Tutorial, ItemClassification, LocationProgressType
+from worlds.AutoWorld import World, WebWorld
+
+
+class InscrypWeb(WebWorld):
+    theme = "dirt"
+
+    guide_en = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the Inscryption Archipelago Multiworld",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["DrBibop"]
+    )
+
+    guide_fr = Tutorial(
+        "Multiworld Setup Guide",
+        "Un guide pour configurer Inscryption Archipelago Multiworld",
+        "Français",
+        "setup_fr.md",
+        "setup/fr",
+        ["Glowbuzz"]
+    )
+
+    tutorials = [guide_en, guide_fr]
+
+    bug_report_page = "https://github.com/DrBibop/Archipelago_Inscryption/issues"
+
+
+class InscryptionWorld(World):
+    """
+    Inscryption is an inky black card-based odyssey that blends the deckbuilding roguelike,
+    escape-room style puzzles, and psychological horror into a blood-laced smoothie.
+    Darker still are the secrets inscrybed upon the cards...
+    """
+    game = "Inscryption"
+    web = InscrypWeb()
+    options_dataclass = InscryptionOptions
+    options: InscryptionOptions
+    all_items = act1_items + act2_items + act3_items + filler_items
+    item_name_to_id = {item['name']: i + base_id for i, item in enumerate(all_items)}
+    all_locations = act1_locations + act2_locations + act3_locations
+    location_name_to_id = {location: i + base_id for i, location in enumerate(all_locations)}
+
+    def create_item(self, name: str) -> Item:
+        item_id = self.item_name_to_id[name]
+        item_data = self.all_items[item_id - base_id]
+        return InscryptionItem(name, item_data['classification'], item_id, self.player)
+
+    def create_items(self) -> None:
+        nb_items_added = 0
+        useful_items = (act1_items + act2_items + act3_items) if self.options.goal != Goal.option_first_act \
+            else act1_items
+
+        if self.options.goal != Goal.option_first_act:
+            if self.options.epitaph_pieces_randomization.value == EpitaphPiecesRandomization.option_all_pieces:
+                useful_items.remove(act2_items[3])
+            elif self.options.epitaph_pieces_randomization.value == EpitaphPiecesRandomization.option_in_groups:
+                useful_items.remove(act2_items[2])
+                useful_items[len(act1_items) + 2]['count'] = 3
+            else:
+                useful_items.remove(act2_items[2])
+                useful_items[len(act1_items) + 2]['count'] = 1
+
+        for item in useful_items:
+            for _ in range(item["count"]):
+                new_item = self.create_item(item['name'])
+                self.multiworld.itempool.append(new_item)
+                nb_items_added += 1
+
+        filler_count = len(self.all_locations if self.options.goal != Goal.option_first_act else act1_locations)
+        filler_count -= nb_items_added
+
+        for i in range(filler_count):
```

Comment:
Is this intended to work where if less filler items need to be added than the list of filler items has, it _always_ picks the exact same set?

Reply:
The only situation where that happens is when the goal is Act 1 only, which would only have room for 1 filler item. Are you saying we should be picking randomly from the list of fillers?

Reply:
I'm just saying it seems like a strange decision to me. If you're basing the fact that this works off of how your *current* options work, then you might as well just add filler_count // 2 of one filler and filler_count // 2 of the other (and then add one as needed)

Reply:
Yeah which is why I brought up my initial comment that the last set isn't picked randomly, but if you don't see that as a problem, I'd understand that. Just wanted to make sure you were aware / it was intended